enhancements/microshift/low-latency-workloads-on-microshift.md

```diff
@@ -0,0 +1,612 @@
+reservedMemory:
+- limits:
+    memory: 1100Mi
+  numaNode: 0
+kubeReserved:
+  memory: 500Mi
+systemReserved:
+  memory: 500Mi
+evictionHard:
+  imagefs.available: 15%
+  memory.available: 100Mi
+  nodefs.available: 10%
+  nodefs.inodesFree: 5%
+evictionPressureTransitionPeriod: 0s
+```
+
+#### Extra manifests
+
+Connects Pod's `.spec.runtimeClassName` to CRI-O's runtime.
+If Pod has `.spec.runtimeClassName: microshift-low-latency`,
+it can use annotations specified in CRI-O config with `crio.runtime.runtimes.high-performance`.
+
+```yaml
+apiVersion: node.k8s.io/v1
+handler: high-performance
+kind: RuntimeClass
+metadata:
+  name: microshift-low-latency
+```
+
+
+### Risks and Mitigations
+
+Biggest risk is system misconfiguration.
+It is not known to author of the enhancement if there are configurations (like kernel params, sysctl, etc.)
+that could brick the device, though it seems rather unlikely.
+Even if kernel panic occurs after staging a deployment with new configuration,
+thanks to greenboot functionality within the grub itself, the system will eventually rollback to
+previous deployment.
+Also, it is assumed that users are not pushing new image to production devices without prior verification on reference device.
+
+It may happen that some users need to use TuneD plugins that are not handled by the profile we'll create.
+In such case we may investigate if it's something generic enough to include, or we can instruct them
+to create new profile that would include `microshift-low-latency` profile.
+
+Systemd daemon we'll provide to enable TuneD profile should have a strict requirement before it
+reboots the node, so it doesn't put it into a boot loop.
+This pattern of reboot after booting affects the number of "effective" greenboot retries,
+so customers might need to account for that by increasing the number of retries.
+
+
+### Drawbacks
+
+Approach described in this enhancement does not provide much of the NTO's functionality
+due to the "static" nature of RPMs and packaged files (compared to NTO's dynamic templating),
+but it must be noted that NTO is going beyond low latency.
+
+One of the NTO's strengths is that it can create systemd units for runtime configuration
+(such as offlining CPUs, setting hugepages per NUMA node, clearing IRQ balance banned CPUs,
+setting RPS masks). Such dynamic actions are beyond capabilities of static files shipped via RPM.
+If such features are required by users, we could ship such systemd units and they could be no-op
+unless they're turned on in MicroShift's config. However, it is unknown to author of the enhancement
+if these are integral part of the low latency.
+
+## Open Questions [optional]
+
+- Verify if osbuild blueprint can override a file from RPM 
+  (variables.conf needs to exist for tuned profile, so it's nice to have some fallback)?
+- NTO runs tuned in non-daemon one shot mode using systemd unit.
+  Should we try doing the same or we want the tuned daemon to run continuously?
```

Comment:
I think we might want to have microshift-tuned.service running as a prereq of microshift.service.

Reply:
`microshift-tuned.service` is not `tuned.service` - this point discusses `tuned.service`